app/services/service_provider_updater.rb

```diff
@@ -10,11 +10,11 @@ def run
   def update_local_caches(service_provider)
     issuer = service_provider['issuer']
     if service_provider['active'] == true
-      SERVICE_PROVIDERS[issuer] = service_provider
-      VALID_SERVICE_PROVIDERS << issuer
+      ServiceProvider.find_or_create_by!(issuer: issuer) do |sp|
+        sp.attributes = service_provider
```

Comment:
valid point. How about a `PROTECTED_ATTRIBUTES` constant for
```rb
:id
:created_at
:updated-at
```
since those are all meta that refers to the local record.